### DISCUSSION TITLE
How to preserve escape characters in Theory string args

### DISCUSSION THREAD

**Discussion by mwpowellhtx** [xunit/xunit#2962] [Question]
I have a test case being yielded that looks something like this. Literally the only thing I do with arguments is to array-ify them, i.e. `[..]`.

```csharp
yield return GetDatum("Crazy wild raw whitespaces work", " 1\t 2  \t  3", 1m, 2m, 3m);
```

Literally, I need for this to be preserved: `[' ', '1', '\t', ' ', '2', ' ', ' ', '\t', ' ', ' ', '3']`. As far as I know, on the source code side of things, it is that way.

However, after the string value falls through the theory data, somehow the escape has been char-ified, if that makes sense: `[' ', '1', '\', 't', ' ', '2', ' ', ' ', '\', 't', ' ', ' ', '3']`. Starting to see the problem?

This makes a huge difference, of course, since I am testing a parser for actual whitespace scenarios, and not for a backslash characters.

Generally that would go for any escape character sequence, I think, but in particular for the [_horizontal tab_ (`0x09`)](https://ascii-code.com/character/%E2%90%89), in this case.

So my dilemma is how to preserve that. Do I break up the string party to constituent char array instead, and reassemble in the theory method? That could work, but moreover, I want to understand, is this a bug in xunit perhaps, how better/otherwise to workaround it, other than like I said, wondering about to its char array.

Thoughts?

**Comment by mwpowellhtx** [comment]
I took a step back and decided to drop the horizontal tab from this particular specification after all, and in favor of vanilla space separation.

**Reply by bradwilson** [reply]
I'm not sure what you mean:

![image](https://github.com/xunit/xunit/assets/16944/40e3b815-db46-43f5-913e-3c00b36f11ca)
